game/cell.py

```diff
@@ -63,12 +63,30 @@ def _middle_char(self, representation: list):
             representation[2] = 'F'
 
     def _put_treasure(self, representation: list):
-        if self.gold:
+        if self.there_is_gold():
             representation[1] = str(self.gold)
-        if self.diamond:
+        if self.there_is_diamond():
             representation[-2] = 'D'
 
     def remove_character(self) -> None:
         player = self.character.player
         player.characters.remove(self.character)
         self.character = None
+
+    def there_is_gold(self):
+        return (self.gold or self.character.gold
+                if self.character
+                else self.gold)
+
+    def there_is_diamond(self):
+        return (self.diamond or self.character.diamond
+                if self.character
+                else self.diamond)
+
+    @TreasureHolder.gold.getter
+    def gold(self):
+        return super().gold + self.character.gold if self.character else super().gold
```

Comment:
I feel this expression conveys the meaning better
```suggestion
        return super().gold + (self.character.gold if self.character else 0)
```

Reply:
Yes, It's more readable and elegant. Thank you